### PR TITLE
Fix ARM64 build with Visual C++.

### PR DESCRIPTION
**Pull request by mtl1979**
* Correct header is arm64_neon.h, not arm_neon.h
* functable.c was missing disabling of NEON check, Visual C++ does not have getauxval()

### CODE REVIEW

**Review by nmoinvaz — COMMENTED** [review]
The indentation for # should be two spaces, but in these changes it is only one space.

File: arch/arm/adler32_neon.c [hunk 1]
```diff
@@ -18,7 +18,11 @@
  */
 #include "adler32_neon.h"
 #if defined(__ARM_NEON__) || defined(__ARM_NEON)
-#include <arm_neon.h>
+#ifdef _M_ARM64
+# include <arm64_neon.h>
```
Comment (nmoinvaz) [inline arch/arm/adler32_neon.c]: For example, should be two spaces between # and include.

**Comment by nmoinvaz** [comment]
Why does `ARM_NOCHECK_NEON` exist?

File: arch/arm/adler32_neon.c (hunk 1 above)
Comment (mtl1979) [reply arch/arm/adler32_neon.c]: Indentation in these files doesn't conform to our own guidelines yet... Whole file needs to be reformatted... I can change the 1 space indentation to 2 space indentation, but that doesn't completely fix the formatting...

**Comment by mtl1979** [comment]
@nmoinvaz Just like we don't check for SSE2 on x86_64, we don't need to check for NEON on ARM64 because it's always available when compiling with Visual C++, it's mandatory requirement.

File: functable.c [hunk 2]
```diff
@@ -121,7 +121,9 @@ ZLIB_INTERNAL uint32_t adler32_stub(uint32_t adler, const unsigned char *buf, si
     functable.adler32 = &adler32_c;
 
 #if (defined(__ARM_NEON__) || defined(__ARM_NEON)) && defined(ARM_NEON_ADLER32)
+# ifndef ARM_NOCHECK_NEON
```
Comment (nmoinvaz) [inline functable.c]: This should be two spaces instead of one, same with the #endif.

**Review by nmoinvaz — APPROVED** [review]
Looks good. I am wondering if it is possible to add a CI build configuration for MSVC ARM64. What commands are using to do the build?

**Comment by mtl1979** [comment]
```vcvarsamd64_arm64.bat``` and ```nmake -f win32\Makefile.a64```

**Comment by Dead2** [comment]
Needs rebase

**Comment by nmoinvaz** [comment]
Looks like it has been rebased.

**Comment by mtl1979** [comment]
@nmoinvaz Yeah... I rebased but Codecov was giving weird errors so I waited if it would fix itself before making a new comment...

**Pull request merged**